lib/helpers.js

```diff
@@ -23,6 +23,28 @@ function createHeader({input, nonce}) {
   // append nonce onto input to create `header` to hash
   const header = Buffer.allocUnsafe(constants.HEADER_SIZE).fill(0);
   input.copy(header, 0);
-  header.writeInt32LE(nonce, constants.HEADER_SIZE - constants.NONCE_SIZE);
+  header.writeUInt32LE(nonce, constants.HEADER_SIZE - constants.NONCE_SIZE);
   return header;
 }
+
+/**
+ * Check if input difficulty is above a set value.
+ *
+ * @param edges an array of unsigned 32 bit integers representing the edges list.
+ * @param difficulty a 32 byte Node.js Buffer representing a target difficulty
+ *          (0 < T < 2^256) that the blake2b hash of the sorted edgelist
+ *          (ascending and serialized little endian) must be less than.
+ *
+ * @return true if meets difficuly threshold, else false.
+ */
+api.checkDifficulty = ({edges, difficulty}) => {
+  // FIXME: check if sorted, always sort, or assume edges are sorted?
+  //const sortedEdges = Array.from(edges).sort();
+  const sortedEdges = edges;
+  const edgesBuf = Buffer.allocUnsafe(sortedEdges.length * 4);
+  for(let i = 0; i < sortedEdges.length; ++i) {
+    edgesBuf.writeUInt32LE(sortedEdges[i], i * 4);
+  }
+  const edgesHash = chloride.crypto_generichash(edgesBuf, constants.HASH_SIZE);
```

Comment:
Can we get blake2b from another lib? `chloride` isn't very actively maintained and we're looking to drop it elsewhere and replace with [sodium-universal](https://github.com/sodium-friends/sodium-universal) or [sodium-native](https://github.com/sodium-friends/sodium-native).